### PR TITLE
🧪 Gather coverage @ CI and upload to Codecov

### PR DESCRIPTION
**Pull request by webknjaz**
##### SUMMARY

$sbj. On one of that calls, people weren't sure if coverage is collected. In this patch, I explore if it's possible to upload it to Codecov.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other

##### AWX VERSION

```
devel
```

##### ADDITIONAL INFORMATION

Coverage view for this PR: https://app.codecov.io/github/ansible/awx/commit/c65873109ac209ec88a298910c5c951497dac4bc/tree/awx.

Installing https://github.com/apps/codecov/installations/398480 into this repo, will enable Codecov to also report status checks.

### CODE REVIEW

**Comment by webknjaz** [comment]
https://app.codecov.io/github/ansible/awx/commit/c65873109ac209ec88a298910c5c951497dac4bc/tree/awx

**Comment by webknjaz** [comment]
Turned this into draft, having seen that `make test` and `make test_coverage` don't run equivalent commands.

**Comment by webknjaz** [comment]
So some Git paleontology led me to https://github.com/ansible/awx/commit/3d45806#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R370 which suggests that originally `make test` and `make test_coverage` were supposed to be the same except for the structured reports for coverage + test run being produced. But they drifted apart over time.

To bring the invocations closer together, I've decided to call one from the other.

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+
+codecov:
+  notify:
+    after_n_builds: 1  # Number of test matrix+lint jobs uploading coverage
+    wait_for_ci: false
+
+  require_ci_to_pass: false
+
+  token: 2b8c7a7a-7293-4a00-bf02-19bd55a1389b  # repo-scoped
```
Comment (TheRealHaoLiu) [inline .codecov.yml]: please add comment around why is this token in plain text and the scope and permission of this token

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ env:
   CI_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   DEV_DOCKER_OWNER: ${{ github.repository_owner }}
   COMPOSE_TAG: ${{ github.base_ref || 'devel' }}
+  UPSTREAM_REPOSITORY_ID: 91594105
```
Comment (TheRealHaoLiu) [inline .github/workflows/ci.yml:8]: please add comment around where the repo id was obtained from

**Review by TheRealHaoLiu — APPROVED** [review]

**Pull request merged**